src/storage/txn/commands.rs

```diff
@@ -1,60 +1,212 @@
+                req.get_txn_size(),
+                req.get_min_commit_ts().into(),
+                req.take_context(),
+            )
+        } else {
+            let is_pessimistic_lock = req.take_is_pessimistic_lock();
+            let mutations = req
+                .take_mutations()
+                .into_iter()
+                .map(Into::into)
+                .zip(is_pessimistic_lock.into_iter())
+                .collect();
+            PrewritePessimistic::new(
+                mutations,
+                req.take_primary_lock(),
+                req.get_start_version().into(),
+                req.get_lock_ttl(),
+                for_update_ts.into(),
+                req.get_txn_size(),
+                req.get_min_commit_ts().into(),
+                req.take_context(),
+            )
         }
     }
+}
 
-    pub fn from_raw_get(request: &mut RawGetRequest) -> Self {
-        PointGetCommand {
-            ctx: request.take_context(),
-            key: Key::from_raw(request.get_key()),
-            ts: None,
-        }
+impl From<PessimisticLockRequest> for Command {
+    fn from(mut req: PessimisticLockRequest) -> Command {
+        let keys = req
+            .take_mutations()
+            .into_iter()
+            .map(|x| match x.get_op() {
+                Op::PessimisticLock => (
+                    Key::from_raw(x.get_key()),
+                    x.get_assertion() == Assertion::NotExist,
+                ),
+                _ => panic!("mismatch Op in pessimistic lock mutations"),
+            })
+            .collect();
+
+        AcquirePessimisticLock::new(
+            keys,
+            req.take_primary_lock(),
+            req.get_start_version().into(),
+            req.get_lock_ttl(),
+            req.get_is_first_lock(),
+            req.get_for_update_ts().into(),
+            WaitTimeout::from_encoded(req.get_wait_timeout()),
+            req.take_context(),
+        )
     }
+}
 
-    #[cfg(test)]
-    pub fn from_key_ts(key: Key, ts: Option<TimeStamp>) -> Self {
-        PointGetCommand {
-            ctx: Context::default(),
-            key,
-            ts,
+impl From<CommitRequest> for Command {
+    fn from(mut req: CommitRequest) -> Command {
+        let keys = req.get_keys().iter().map(|x| Key::from_raw(x)).collect();
+
+        Commit::new(
+            keys,
+            req.get_start_version().into(),
+            req.get_commit_version().into(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<CleanupRequest> for Command {
+    fn from(mut req: CleanupRequest) -> Command {
+        Cleanup::new(
+            Key::from_raw(req.get_key()),
+            req.get_start_version().into(),
+            req.get_current_ts().into(),
+            req.take_context(),
+        )
+    }
+}
+
+impl From<BatchRollbackRequest> for Command {
+    fn from(mut req: BatchRollbackRequest) -> Command {
+        let keys = req.get_keys().iter().map(|x| Key::from_raw(x)).collect();
```

Comment:
Please add a blank line behind.

Reply:
Sorry, I'm not clear where you want the blank line. Could you explain why?